lib/samples/filter_building_scene_layer/filter_building_scene_layer.dart

```diff
@@ -0,0 +1,343 @@
+    final statistics = await _buildingSceneLayer.fetchStatistics();
+    if (statistics['BldgLevel'] != null) {
+      final floorList = <String>[];
+      floorList.addAll(statistics['BldgLevel']!.mostFrequentValues);
+      floorList.sort((a, b) => int.parse(b).compareTo(int.parse(a)));
+      setState(() {
+        _floorList = floorList;
+      });
+    }
+
+    // Apply the scene to the local scene view controller.
+    _localSceneViewController.arcGISScene = scene;
+
+    // Set the ready state variable to true to enable the sample UI.
+    setState(() => _ready = true);
+  }
+
+  Future<void> onTap(Offset offset) async {
+    // Clear the current selection
+    if (_selectedSublayer != null) {
+      _selectedSublayer!.clearSelection();
+      _selectedSublayer = null;
+    }
+
+    // Identify on the building scene layer.
+    final identifyResult = await _localSceneViewController.identifyLayer(
+      _buildingSceneLayer,
+      screenPoint: offset,
+      tolerance: 5,
+    );
+
+    // Select the first identified feature and show the feature details in a popup.
+    if (identifyResult.sublayerResults.isNotEmpty) {
+      final sublayerResult = identifyResult.sublayerResults.first;
+      if (sublayerResult.geoElements.isNotEmpty) {
+        final identifiedFeature = sublayerResult.geoElements.first as Feature;
+        final sublayer =
+            sublayerResult.layerContent as BuildingComponentSublayer;
+        sublayer.selectFeature(identifiedFeature);
+        _selectedSublayer = sublayer;
+
+        if (mounted) {
+          showFeatureDetail(context: context, feature: identifiedFeature);
+        }
+      }
+    }
+  }
+
+  // Utility function to update the building filters based on the selected floor.
+  void updateFloorFilters() {
+    if (_selectedFloor == 'All') {
+      // No filtering applied if 'All' floors are selected.
+      _buildingSceneLayer.activeFilter = null;
+      return;
+    }
+
+    // Build a building filter to show the selected floor and an xray view of the floors below.
+    // Floors above the selected floor are not shown at all.
+    final buildingFilter = BuildingFilter(
+      name: 'Floor filter',
+      description: 'Show selected floor and xray filter for lower floors.',
+      blocks: [
+        BuildingFilterBlock(
+          title: 'solid block',
+          whereClause: 'BldgLevel = $_selectedFloor',
+          mode: BuildingSolidFilterMode(),
+        ),
+        BuildingFilterBlock(
+          title: 'xray block',
+          whereClause: 'BldgLevel < $_selectedFloor',
+          mode: BuildingXrayFilterMode(),
+        ),
+      ],
+    );
+
+    // Apply the filter to the building scene layer.
+    _buildingSceneLayer.activeFilter = buildingFilter;
+  }
+
+  // Display the feature details in a Popup view in a modal bottom sheet.
+  void showFeatureDetail({
+    required BuildContext context,
+    required Feature feature,
+  }) {
+    if (!mounted) return;
```

Comment:
`mounted` is getting checked twice. I'd prolly remove it from here. Makes more sense to make it the responsibility of the method that is doing `async` work.